File: week1/주식가격.py
def solution(prices):
    answer = []
    p_len = len(prices)
    
    for i in range(p_len-1):
        count = 0
        for j in range(i+1, p_len):
            count += 1
            if prices[i] > prices[j]:
                break
        answer.append(count)
    
    # 어차피 맨 마지막은 0일테니(비교군 없어서)
    answer.append(0)
    return answer

File: week1/주식가격.java
package codingtest;

import java.util.*;

class Solution {
    public int[] solution(int[] prices) {
        int[] answer = new int [prices.length];
        Deque <Integer> st = new ArrayDeque<>();
        
        
        for(int i =0; i<prices.length; i++) {
        	int cur_price = prices[i];
        	while(!st.isEmpty() && prices[st.peek()]>cur_price) { // 들어오는 가격보다 낮으면.
        		int idx = st.pop();
        		answer[idx] = i -idx;
        	}	
        	st.push(i);

        }
        
        while(!st.isEmpty()){
        	int idx = st.pop();
        	answer[idx] = prices.length - idx-1;
        }
        
        return answer;
    }
}